lib/sla.js

```diff
@@ -128,13 +186,58 @@ function updateState(client, name, state) {
  * @param {Object} inbound- the inbound channel
  * @param {Object} bridge - Object that defines the bridge to be passed to
  *   addChannelsToBridge().
- * @ param {Object} trunkData - the data of the trunk to be used for station
- *   access.
+ * @param {Object} extension - the data structure that represents the extension,
+ *   mainly used for telling whether or not an extension is busy or not
  * @return {Q} Q promise object.
  */
-function originateChannel(client, inbound, bridge, trunkData) {
-  var defer = Q.defer();
+function originateInbound(client, inbound, bridge, extension) {
+
   var dialed = [];
+  var defer = Q.defer();
+
+  getState(client, bridge.name).then(function(currentState) {
+
+    if (currentState !== inUse && currentState !== busy &&
+        currentState !== ringing) {
+      extension.getAllStations()
+        .then(function (stations) {
+
+          if (stations.length) {
+            stations.forEach(function(station) {
+              dialed.push({endpoint: station, channel: client.Channel()});
+            });
+
+            dialed.forEach(function (object) {
+              var originate = Q.denodeify(object.channel.originate.bind(
+                  object.channel));
+              originate({endpoint: object.endpoint, app: 'sla',
+                  appArgs: 'dialed', timeout: 10})
+                .catch(function (err) {
+                  defer.reject(err);
+                });
+
+              updateState(client, bridge.name, ringing);
+
+              object.channel.once('StasisStart', dialedEnteredStasis);
+              object.channel.once('ChannelDestroyed', dialedChannelHungup);
+            });
+           inbound.once('ChannelHangupRequest', inboundHungup);
+          } else {
+            defer.reject(new customError.CustomError('NoStations',
+                  'No stations in this shared extension.'));
+          }
+        });
+    } else {
+      // This will either redirect this channel towards a Hangup or a "backup
+      // extension"
```

Comment:
Add an item to your wiki project plan to set a channel variable if an inbound trunk is bounced out of a shared extension due to it being already in use by another trunk.